gems/pending/appliance_console/date_time_configuration.rb

```diff
@@ -0,0 +1,96 @@
+require "appliance_console/logging"
+require 'appliance_console/prompts'
+
+module ApplianceConsole
+  class DateTimeConfiguration
+    DATE_REGEXP   = /^(2[0-9]{3})-(0?[1-9]|1[0-2])-(0?[1-9]|[12][0-9]|3[01])/
+    DATE_PROMPT   = "current date (YYYY-MM-DD)".freeze
+    TIME_REGEXP   = /^(0?[0-9]|1[0-9]|2[0-3]):([0-5][0-9]):([0-5][0-9])/
+    TIME_PROMPT   = "current time in 24 hour format (HH:MM:SS)".freeze
+
+    attr_accessor :new_date, :new_time
+
+    include ApplianceConsole::Logging
+
+    def initialize(config = {})
+      @new_date = config[:new_date]
+      @new_time = config[:new_time]
+    end
+
+    def activate
+      say("Applying time configuration...")
+      disable_auto_sync &&
+        configure_date_time
+    end
+
+    def ask_questions
+      ask_disable_auto_sync &&
+        ask_for_date &&
+        ask_for_time &&
+        confirm
+    end
+
+    def ask_disable_auto_sync
+      say("Automatic time synchronization must be disabled to manually set date or time\n\n")
+
+      return false unless agree("Disable Automatic time synchronization? (Y/N): ")
+      say("Automatic time synchronization will be disabled...")
+      true
+    end
+
+    def ask_for_date
+      begin
+        @new_date = just_ask(DATE_PROMPT, nil, DATE_REGEXP)
```

Comment:
> new_date = just_ask(DATE_PROMPT, nil, DATE_REGEXP)

you'd need to do `self.new_date =` otherwise it's unclear if it's a local variable or a setter.  But, using an ivar is fine here.